src/nnet3/nnet-test-utils.cc

```diff
@@ -793,7 +793,7 @@ void ComputeExampleComputationRequestSimple(
 
 static void GenerateRandomComponentConfig(std::string *component_type,
                                           std::string *config) {
-  int32 n = RandInt(0, 20);
+  int32 n = RandInt(0, 21);
```

Comment:
Definitly remove the CollapseWithNext() and Resize() functions from this
component.
Regarding removing them from other components-- there may be a time when we
need them, so let's just leave the code there for now.
Dan


On Wed, Nov 4, 2015 at 3:26 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In src/nnet3/nnet-test-utils.cc
> <https://github.com/kaldi-asr/kaldi/pull/307#discussion_r43934452>:
>
> > @@ -793,7 +793,7 @@ void ComputeExampleComputationRequestSimple(
> >
> >  static void GenerateRandomComponentConfig(std::string *component_type,
> >                                            std::string *config) {
> > -  int32 n = RandInt(0, 20);
> > +  int32 n = RandInt(0, 21);
>
> @dpovey <https://github.com/dpovey> May be it would be better to just
> delete these methods from all the components.
> … <#150d42d7881b6231_>
> --Vijay
> On Wed, Nov 4, 2015 at 3:24 PM, hhadian <notifications@github.com> wrote:
> In src/nnet3/nnet-test-utils.cc <#307 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/307#discussion_r43934253>>: > @@
> -793,7 +793,7 @@ void ComputeExampleComputationRequestSimple( > > static
> void GenerateRandomComponentConfig(std::string *component_type, >
> std::string *config) { > - int32 n = RandInt(0, 20); > + int32 n =
> RandInt(0, 21); Yes. Except abs-min which I assumed unnecessary based on
> your comment (and also there was no such function already implemented).
> Also I did not remove CollapseWithNext() and Resize() from ScaleComponent
> because I was not sure if you wanted them to be deleted too. — Reply to
> this email directly or view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/307/files#r43934253>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/307/files#r43934452>.
>